### PR TITLE
CDK-938: Add kite-data-s3

### PR DESCRIPTION
**Pull request by rdblue**
This adds both s3n and s3a dataset URIs, s3n is working with a test.
Both URI patterns assume the bucket is the dataset repository.

Until a mock S3 service can be added, this expects real S3 credentials
and a bucket name passed as S3_ID, S3_KEY, and S3_BUCKET environment
variables. If those are not present, then the test will not run. To
disable tests for the hadoop1 and cdh4 profiles, the credentials are
explicitly set to empty.

### CODE REVIEW

File: kite-data/kite-data-s3/src/test/java/org/kitesdk/data/spi/s3/TestS3Dataset.java [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+import org.kitesdk.data.DatasetDescriptor;
+import org.kitesdk.data.DatasetReader;
+import org.kitesdk.data.DatasetWriter;
+import org.kitesdk.data.Datasets;
+import org.kitesdk.data.spi.DefaultConfiguration;
+
+public class TestS3Dataset {
+  private static final String ID = System.getProperty("aws.s3.id");
+  private static final String KEY = System.getProperty("aws.s3.key");
+  private static final String BUCKET = System.getProperty("aws.s3.bucket");
```
Comment (tomwhite) [inline kite-data/kite-data-s3/src/test/java/org/kitesdk/data/spi/s3/TestS3Dataset.java]: Put "test" in the property names to be clear that these are for testing only. Also, "key" is ambiguous, so spell out "aws.s3.access.key" and "aws.s3.secret.key" (to mirror s3a)?

**Comment by tomwhite** [comment]
Looks good to me, pending one naming point.

File: kite-data/kite-data-s3/src/test/java/org/kitesdk/data/spi/s3/TestS3Dataset.java (hunk 1 above)
Comment (rdblue) [inline kite-data/kite-data-s3/src/test/java/org/kitesdk/data/spi/s3/TestS3Dataset.java]: Will do. Thanks for taking a look!

**Comment by tomwhite** [comment]
I've updated this in https://github.com/kite-sdk/kite/pull/356 with the renames, and also to allow a path in the S3 URI.

**Pull request closed without merging**